1.segment_images/scripts/4.reconstruct_3D_masks.py

```diff
@@ -77,21 +83,27 @@
     compartment = args.compartment
 else:
     print("Running in a notebook")
-    input_dir = pathlib.Path("../processed_data/C6-1/").resolve(strict=True)
-    x_y_vector_radius_max_constaint = 10  # pixels
-    compartment = "nucleui"
-
-mask_path = pathlib.Path(f"../processed_data/{input_dir.stem}").resolve()
-mask_path.mkdir(exist_ok=True, parents=True)
+    input_dir = pathlib.Path("../processed_data/C4-2/").resolve(strict=True)
+    compartment = "organoid"
 
+mask_dir = pathlib.Path(f"../processed_data/{str(input_dir.stem)}").resolve(strict=True)
 if compartment == "nuclei":
-    input_image_dir = pathlib.Path(mask_path / "nuclei_masks.tiff").resolve(strict=True)
-    output_image_dir = pathlib.Path(mask_path / "nuclei_masks.tiff").resolve()
+    input_image_dir = pathlib.Path(mask_dir / "nuclei_masks.tiff").resolve(strict=True)
+    x_y_vector_radius_max_constaint = 5  # pixels
 elif compartment == "cell":
-    input_image_dir = pathlib.Path(mask_path / "cell_masks.tiff").resolve(strict=True)
-    output_image_dir = pathlib.Path(mask_path / "cell_masks.tiff").resolve()
+    input_image_dir = pathlib.Path(mask_dir / "cell_masks.tiff").resolve(strict=True)
+    x_y_vector_radius_max_constaint = 15  # pixels
+elif compartment == "organoid":
+    input_image_dir = pathlib.Path(mask_dir / "organoid_masks.tiff").resolve(
+        strict=True
+    )
+    x_y_vector_radius_max_constaint = 50  # pixels
```

Comment:
I think this will be a similar comment you have gotten before, but specifically for organoids, have you considered not using Euclidian distance with the center x and y coords and instead using the masks themselves with overlap? Since many of these organoids have variable intensity across z-slices, I can see a world where this distance threshold might not be sufficient across a whole organoid. This could relate to the flashing in and out of the organoid we were seeing in the GIFs.